Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentGraph.php

```diff
@@ -187,6 +188,39 @@ public function findParentNodeAggregates(
         return $this->mapQueryBuilderToNodeAggregates($queryBuilder);
     }
 
+    public function findAncestorNodeAggregateIds(NodeAggregateId $entryNodeAggregateId): NodeAggregateIds
+    {
+        $queryBuilderInitial = $this->createQueryBuilder()
+            ->select('n.nodeAggregateId, ch.parentnodeanchor')
+            ->from($this->nodeQueryBuilder->tableNames->node(), 'n')
+            ->innerJoin('n', $this->nodeQueryBuilder->tableNames->hierarchyRelation(), 'ch', 'ch.parentnodeanchor = n.relationanchorpoint')
+            ->innerJoin('ch', $this->nodeQueryBuilder->tableNames->node(), 'c', 'c.relationanchorpoint = ch.childnodeanchor')
+            ->where('ch.contentstreamid = :contentStreamId')
+            ->andWhere('c.nodeaggregateid = :entryNodeAggregateId');
+
+        $queryBuilderRecursive = $this->createQueryBuilder()
+            ->select('pn.nodeAggregateId, h.parentnodeanchor')
+            ->from('ancestry', 'cn')
+            ->innerJoin('cn', $this->nodeQueryBuilder->tableNames->node(), 'pn', 'pn.relationanchorpoint = cn.parentnodeanchor')
```

Comment:
cn should actually be named ch, right? It's joined as the child hierarchy relation
h could then be ph, just for better understanding